File: argentum-api/src/main/java/net/devium/argentum/rest/model/ProductRangeResponseMeta.java
package net.devium.argentum.rest.model;

import net.devium.argentum.jpa.ProductRangeEntity;

public class ProductRangeResponseMeta {
    private long id;
    private String name;

    public ProductRangeResponseMeta() {
    }

    public ProductRangeResponseMeta(long id, String name) {
        this.id = id;
        this.name = name;
    }

    public static ProductRangeResponseMeta from(ProductRangeEntity range) {
        return new ProductRangeResponseMeta(
                range.getId(),
                range.getName()
        );
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

File: argentum-ui/src/app/common/rest-service/response/product-range-response-meta.ts
import { ProductRange } from '../../model/product-range';
export class ProductRangeResponseMeta {
  id: number;
  name: string;
}

export function toProductRangeMeta(response: ProductRangeResponseMeta): ProductRange {
  return {
    id: response.id,
    name: response.name,
    products: null
  };
}
